src/edu/stanford/nlp/sempre/YearRangeFn.java

```diff
@@ -0,0 +1,60 @@
+package edu.stanford.nlp.sempre;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parse year ranges of the form "1990's" or "1800s"
+ *
+ * @author ppasupat
+ */
+public class YearRangeFn extends SemanticFn {
```

Comment:
I agree. We can certainly extend this to other date ranges.

One difficulty though is that we are depending on CoreNLP's NER, which unfortunately doesn't include all possible interpretation of the input (e.g., "1900's" could be a decade or a century, depending on the context). This class is a small hack that returns all possible interpretations.